go/vt/tabletserver/grpctabletconn/conn.go

```diff
@@ -139,50 +138,51 @@ func (conn *gRPCQueryClient) ExecuteBatch(ctx context.Context, target *querypb.T
 	return sqltypes.Proto3ToResults(ebr.Results), nil
 }
 
-type streamExecuteAdapter struct {
-	stream queryservicepb.Query_StreamExecuteClient
-	fields []*querypb.Field
-}
+// StreamExecute starts a streaming query to VTTablet.
+func (conn *gRPCQueryClient) StreamExecute(ctx context.Context, target *querypb.Target, query string, bindVars map[string]interface{}, options *querypb.ExecuteOptions, sendReply func(*sqltypes.Result) error) error {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
 
-func (a *streamExecuteAdapter) Recv() (*sqltypes.Result, error) {
-	ser, err := a.stream.Recv()
-	switch err {
-	case nil:
-		if a.fields == nil {
-			a.fields = ser.Result.Fields
+	stream, err := func() (queryservicepb.Query_StreamExecuteClient, error) {
```

Comment:
It's for the defer. This pattern is now in all four streaming functions. I'm looking for ways to avoid this boiler-plating.